cognite/neat/_graph/transformers/_prune_graph.py

```diff
@@ -1,84 +1,131 @@
-                                     ?destinationNode <{property_predicate}> ?{property_name} . }}"""
+
+    _query_template_keep_old_predicate: str = """
+    SELECT ?sourceNode ?property_source ?destinationNode ?property_destination ?property_value WHERE {{
+        ?sourceNode ?property_source ?destinationNode .
+        ?destinationNode ?property_destination ?property_value .
+        ?destinationNode a <{destination_node_type}> .
+        ?destinationNode <{value_property_predicate}> ?property_value . }}"""
+
+    _query_template_new_predicate: str = """
+    SELECT ?sourceNode ?property ?destinationNode ?predicate_value ?property_value WHERE {{
+        ?sourceNode ?property ?destinationNode .
+        ?destinationNode a <{destination_node_type}> .
+        ?destinationNode <{predicate_property_predicate}> ?predicate_value .
+        ?destinationNode <{value_property_predicate}> ?property_value . }}"""
 
     def __init__(
         self,
         destination_node_type: URIRef,
-        property_predicate: Namespace,
-        property_name: str,
-        delete_connecting_node: bool = True,
+        namespace: Namespace,
+        value_property_name: str,
+        delete_connecting_node: bool = False,
+        predicate_property_name: str | None = None,
     ):
         self.destination_node_type = destination_node_type
-        self.property_predicate = property_predicate
-        self.property_name = property_name
+        self.namespace = namespace
+        self.value_property_predicate = self.namespace[value_property_name]
         self.delete_connecting_node = delete_connecting_node
+        self.predicate_property_name = predicate_property_name
 
-    def transform(self, graph: Graph) -> None:
-        nodes_to_delete: list[Identifier] = []
-
-        graph_traversals = list(
-            graph.query(
-                self._query_template.format(
-                    destination_node_type=self.destination_node_type,
-                    property_predicate=self.property_predicate,
-                    property_name=self.property_name,
-                )
+    def transform(self, graph) -> None:
+        nodes_to_delete: list = []
+
+        if self.predicate_property_name is not None:
+            predicate_property_predicate = self.namespace[self.predicate_property_name]
+            query = self._query_template_new_predicate.format(
+                destination_node_type=self.destination_node_type,
+                predicate_property_predicate=predicate_property_predicate,
+                value_property_predicate=self.value_property_predicate,
+            )
+        else:
+            query = self._query_template_keep_old_predicate.format(
+                destination_node_type=self.destination_node_type,
+                value_property_predicate=self.value_property_predicate,
+            )
+
+        graph_traversals = graph.query(query)
+
+        for result_raw in graph_traversals:
```

Comment:
```suggestion
        for (source_node, predicate, destination_node, property_value,) in graph.query(query):
```

you can already here unpack results